flipperdevices/flipperzero-firmware
# Improve vscode clangd experience

# What's new

- Resolve and use absolute path for the toolchain. This allows clangd to use `compile_commands.json` file without running under fbtenv, simplifies setup for vscode clangd extension (AFAIK clangd solely relies on PATH to look for the compiler drivers instead of having a way to specify the location like cpptools. There seems to be an option to specify a compiler override but it seems to be related to `compile_flags.txt` only and doesn't work for this use case). As a side effect, a rebuild is needed to update `compile_commands.json` after moving the source tree but it shouldn't cause much trouble in practice.
- Add the recommended default settings of the extension to `settings.json`.

# Verification 

- (If have `ms-vscode.cpptools`) Disable `ms-vscode.cpptools`.
- Install `llvm-vs-code-extensions.vscode-clangd`.
- Run `./fbt` and `./fbt vscode_dist LANG_SERVER=clangd`, then open the project in vscode.
- Navigate to `furi/flipper.c` and other files.
- Everything should be correctly indexed including the libc headers.

# Checklist (For Reviewer)
